source/github-helpers/is-default-branch.ts

```diff
@@ -24,5 +24,5 @@ export default async function isDefaultBranch(): Promise<boolean> {
 	// defaultBranch === 'a/b' && currentBranch === 'a'
 	const path = parts.join('/');
 	const defaultBranch = await getDefaultBranch();
-	return path === defaultBranch || defaultBranch.startsWith(`${path}/`);
+	return path === defaultBranch || path.startsWith(`${defaultBranch}/`);
```

Comment:
Scenarios:

- path is `main`
- path is `main/dir/file`

So:

- `'main' === 'main'`
- `'main/dir/file'.startsWith('main/')`